server/handlers/component_handler.go

```diff
@@ -1134,13 +1140,18 @@ func (h *Handler) GetAllMeshmodelComponents(rw http.ResponseWriter, r *http.Requ
 	entities, count, _ := h.registryManager.GetEntities(filter)
 	var comps []v1alpha1.ComponentDefinition
 	for _, r := range entities {
+		host := h.registryManager.GetRegistrant(r)
+
 		comp, ok := r.(v1alpha1.ComponentDefinition)
 		if ok {
 			m := make(map[string]interface{})
 			_ = json.Unmarshal([]byte(comp.Schema), &m)
 			m = core.Format.Prettify(m, true)
 			b, _ := json.Marshal(m)
 			comp.Schema = string(b)
+			comp.HostID = host.ID
+			comp.HostName = host.Hostname
```

Comment:
@Freedisch Is this still required, since you are already doing it here. https://github.com/meshery/meshkit/pull/337/files#diff-2ff4deb53927cd3792edbed666384a10774880ce34a164439a027d0044dd6729R250